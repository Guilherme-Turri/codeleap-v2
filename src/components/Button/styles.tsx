import styled, { css } from 'styled-components'

export const Button = styled.button`
${({ theme }) => css`
all:unset;
background-color: ${theme.color.secondaryColor};
border-radius:  ${theme.radius.small};
width: 100%;
height: 100%;
display: flex;
justify-content: center;
align-items: center;
cursor: pointer;
`}
`
