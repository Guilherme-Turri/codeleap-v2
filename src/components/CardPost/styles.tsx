import styled, { css } from 'styled-components'
import { isMobile } from '../../types/types'

export const Container = styled.main<isMobile>`
${({ theme, isMobile }) => css`
min-height: 150px;
display: flex;
flex-direction:${isMobile ? 'column' : 'row'} ;
flex: 0 0 85%; 
justify-content: space-evenly;
margin-top: ${theme.spacings.xsmall};
`}
`
export const User = styled.div`
${({ theme }) => css`
display: flex;
flex-direction: column;
justify-content: space-evenly;
align-items: center;
flex: 0 20%; 
a{
  color: ${theme.color.optionalColor};
  font-weight: bold;
  cursor: pointer;
}
`}
`
export const Img = styled.img`
${({ theme }) => css`
width: 5.625rem;
height: 5.625rem;
display: flex;
border-radius: ${theme.radius.total};
object-fit: cover;
`}
`
export const Post = styled.div`
${({ theme }) => css`
display: flex;
flex: 0 80%; 
flex-direction: column;
justify-content:space-around;
padding: ${theme.spacings.small};
& > *:nth-child(1) {
  color: ${theme.color.optionalColor};
  font-weight: bold;
}
 & > *:nth-child(3) {
  font-size: ${theme.sizes.small};
  `}
`
