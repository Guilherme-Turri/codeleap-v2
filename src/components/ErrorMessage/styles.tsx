import styled, {css} from 'styled-components'

export const Container = styled.p`
${({theme}) => css`
color: ${theme.color.error};
font-size: ${theme.sizes.small};
`}
`