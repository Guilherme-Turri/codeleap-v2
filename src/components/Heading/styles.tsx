import styled, { css } from 'styled-components'
import { HeadingProps } from './index'

export const Container = styled.h1<HeadingProps>`
  ${({ theme, uppercase, big }) => css`
      text-transform: ${uppercase ? 'uppercase' : 'none'};
      color: ${theme.color.secondaryColor};
      font-size: ${big ? theme.sizes.huge : theme.sizes.xbig};
      font-family: ${theme.fonts.logoFont};
      `}
  `