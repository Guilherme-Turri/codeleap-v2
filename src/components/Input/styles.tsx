import styled, { css } from 'styled-components'


export const Container = styled.input`
${({ theme }) => css`
outline: none;
all: unset;
border-bottom: 1px solid ${theme.color.secondaryColor};
width: 100%;
padding: ${theme.spacings.small};
box-sizing: border-box;
color: ${theme.color.primaryColor};
::placeholder{
  color:'#95A5A6'!important;
}
`}
`