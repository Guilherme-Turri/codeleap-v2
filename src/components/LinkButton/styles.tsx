import styled, { css } from 'styled-components'
interface ContainerProps {
  size: 'small' | 'medium' | 'large'
}

const setWidth = {
  small: () => css`
  width: 20%;
  `,
  medium: () => css`
  width:50%;
  `,
  large: () => css`
  width:70%;
  `,
}

export const Container = styled.div<ContainerProps>`
${({ theme, size }) => css`
display:flex;
width:100%;
justify-content:center;
  a{
  display:flex;
  ${setWidth[size]}
  background:${theme.color.secondaryColor};
  color: white;
  padding: ${theme.spacings.xsmall};
  border-radius: ${theme.radius.small};
  justify-content: center;
  align-items: center;
  }
`}
`