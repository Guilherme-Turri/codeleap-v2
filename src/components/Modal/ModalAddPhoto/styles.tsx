import styled, { css } from 'styled-components'
import { Container as TC } from '../../TextComponent/styles'
interface BtnProps {
  loadPhoto?: boolean
}

export const Container = styled.div`
${({ theme }) => css`
display: flex;
justify-content: space-around;
align-items: center;
flex-direction: column;
width: 100%;
height: 100%;
padding: ${theme.spacings.xsmall};;
${TC}{
  font-size: 1.5rem;
}
input{
  all: unset;
}
`}
`

export const Button = styled.div<BtnProps>`
${({ theme, loadPhoto }) => css`
display:${loadPhoto ? 'none' : 'flex'};
justify-content: space-evenly;
align-items: center;
width: 6rem;
height: 3rem;
margin:  ${theme.spacings.normal};
`}
`

export const Wrapper = styled.div`
display:flex;
align-items: center;
width: 100%;
justify-content: space-evenly;
`

export const Close = styled.div`
${({ theme }) => css`
  position:absolute;
  display:flex;
  width:3.75rem;
  height: 3.75rem;
  border-radius: ${theme.radius.total};
  top:-20px;
  right:-20px;
  cursor:pointer;
`}
`

