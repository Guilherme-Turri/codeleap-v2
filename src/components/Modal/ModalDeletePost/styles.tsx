import styled, { css } from 'styled-components'
import { Container as TC } from '../../TextComponent/styles'

export const Container = styled.div`
${({ theme }) => css`
display: flex;
justify-content: space-evenly;
align-items: center;
flex-direction: column;
width: 100%;
height: 100%;
${TC}{
  font-size: ${theme.sizes.big};
}
`}
`
export const Button = styled.div`
display:flex;
justify-content: space-evenly;
align-items: center;
width: 6rem;
height: 3rem;
`
export const Wrapper = styled.div`
display:flex;
 flex-direction: column;
align-items: center;
width: 100%;
justify-content: space-evenly; 
`
export const Actions = styled.div`
${({ theme }) => css`
display:flex;
 flex-direction: row;
align-items: center;
width: 100%;
justify-content: space-evenly; 
padding: ${theme.spacings.normal};
`}
`