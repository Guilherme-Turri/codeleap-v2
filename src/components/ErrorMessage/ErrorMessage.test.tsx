import { componentRender } from '../../render/componentRender'
import { ErrorMessage } from '.'
import { screen } from '@testing-library/react'
import { theme } from '../../styles/theme'

describe('<ErrorMessage/>', () => {
  it('should render on screen', () => {
    const { container } = componentRender(<ErrorMessage>test</ErrorMessage>)
    expect(screen.getByText('test')).toBeInTheDocument()
    expect(screen.getByText('test')).toHaveStyle(`color:${theme.color.error}`)
    expect(container).toMatchSnapshot();
  })
})