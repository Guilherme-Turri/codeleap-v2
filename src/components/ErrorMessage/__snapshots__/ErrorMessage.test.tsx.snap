// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage/> should render on screen 1`] = `
.c0 {
  color: #ff5500;
  font-size: 0.85rem;
}

<div>
  <p
    class="c0"
  >
    test
  </p>
</div>
`;
